drivers/UARTSerial.cpp

```diff
@@ -305,11 +305,11 @@ void UARTSerial::rx_irq(void)
 void UARTSerial::tx_irq(void)
 {
     bool was_full = _txbuf.full();
+    char data = 0;
```

Comment:
Isn't this now just masking the problem, which the compiler correctly warned for: if "CricularBuffer::pop()" returns false, the data -variable should not be forwarded to SerialBase::_base_putc()?